x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/policy/AutoscalingPolicy.java

```diff
@@ -48,7 +48,7 @@
             return new AutoscalingPolicy(
                 name,
                 roles.stream().collect(Sets.toUnmodifiableSortedSet()),
-                new TreeMap<>(deciders.stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)))
+                deciders.stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (a, b) -> b, TreeMap::new))
```

Comment:
I think there is a trade-off here in that this now ignores non-unique input. As a reader, the `(a, b) -> b` is puzzling. It is sad that there is no `Collectors.toMap` where you can reuse the `uniqKeysMapMerger` or only specify the map-factory.

Given this I would prefer to keep this as is since it is not entirely obvious from the code that there cannot be duplicates. Or we could add our own utility `toSortedMap` similar to `Sets.toSortedSet` (I think we only rely on it being sorted most of the time).

Reply:
Good point aboit ignoring duplicates. I will add the `toSortedMap` utility!